internal/pkg/service/stream/storage/statistics/repository/stats_delete.go

```diff
@@ -0,0 +1,71 @@
+package repository
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/statistics"
+)
+
+func (r *Repository) rollupStatisticsOnFileDelete() {
+	r.plugins.Collection().OnFileDelete(func(ctx context.Context, now time.Time, original, updated *model.File) error {
+		op.AtomicOpFromCtx(ctx).AddFrom(r.deleteOrRollup(updated.FileKey))
+		return nil
+	})
+}
+
+// deleteOrRollup returns an etcd operation to delete all statistics associated with the object key.
+// Statistics for the level.Target are not deleted but are rolled up to the parent object.
+func (r *Repository) deleteOrRollup(objectKey fmt.Stringer) *op.AtomicOp[op.NoResult] {
+	ops := op.Atomic(r.client, &op.NoResult{})
+	for _, inLevel := range level.AllLevels() {
+		// Object prefix contains all statistics related to the object
+		objectPfx := r.schema.InLevel(inLevel).InObject(objectKey)
+
+		// Keep statistics about successfully imported data in the parent object prefix, in the sum key
+		if inLevel == level.Target {
```

Comment:
Could we do 
```
if inLevel != level.Target { 
ops.Write(func(ctx context.Context) op.Op {
	return objectPfx.DeleteAll(r.client)
})
}
rest of the code ...
```
The code is too nested and this part could be easily evaluated upfront and reduce nesting.

Reply:
Ok

Reply:
[fix: Review fixes](https://github.com/keboola/keboola-as-code/pull/1820/commits/03b95f2f3ba09bd1292215856802fa207b2b71aa)

Reply:
first fix was bad, fix: https://github.com/keboola/keboola-as-code/pull/1820/commits/497f0487ead3d4954616b5b61075f00c4a7fa5ce